src/core/util.cpp

```diff
@@ -754,31 +754,47 @@ namespace mamba
     LockFile::LockFile(const fs::path& path)
         : m_path(path)
     {
-        LOG_INFO << "Locking " << path;
+        if (!fs::exists(path))
+        {
+            LOG_DEBUG << "Could not lock non existing path '" << path.string() << "'";
+            m_fd = -1;
+            return;
+        }
 
-        m_fd = open(path.c_str(), O_RDWR | O_CREAT, 0666);
+        LOG_DEBUG << "Locking '" << path.string() << "'";
+        if (fs::is_directory(path))
+            m_lock = m_path / "mamba.lock";
+        else
```

Comment:
does this really work? because we're checking the existence of the file before hand ... but then we're also trying to create the file in the next step.

The logic seems not entirely correct.

Reply:
don't totally get it, I'll have a look